### PR TITLE
[ONNX] Addition of diagnostic tool API

### PR DESCRIPTION
**Pull request by shubhambhokare1**
Added initial diagnostic tool API

### CODE REVIEW

**Comment by BowenBao** [comment]
Please check and fix flake8 Lint error found by the CI. Please also rebase with master, some of the caffe2 CI tests are failing because of that.

File: torch/onnx/utils.py [hunk 1]
```diff
@@ -483,6 +483,26 @@ def _export_to_pretty_string(model, args, f, export_params=True, verbose=False,
...
+    
+    with select_model_mode_for_export(model, training):
+        graph, params_dict, torch_out = _model_to_graph(model, args, verbose, input_names,
+                                                        output_names, operator_export_type)
+    unsupported_ops = list()
+    
+    for node in graph.nodes():
+        if node.kind().split(':')[0] not in ['onnx', 'prim']:
+            unsupported_ops.append(node.kind())
+    return (graph, unsupported_ops, torch_out)
```
Comment (BowenBao) [inline torch/onnx/utils.py:505]: nit: remove parentheses

File: torch/onnx/utils.py [hunk 2]
```diff
@@ -483,6 +483,26 @@ def _export_to_pretty_string(model, args, f, export_params=True, verbose=False,
                                         operator_export_type, google_printer,
                                         val_keep_init_as_ip, custom_opsets, val_add_node_names)
 
+def _diagnose_export(model, args, f, verbose=False, training=TrainingMode.EVAL,
```
Comment (BowenBao) [inline torch/onnx/utils.py:486]: Please add comments explaining the api, similarly to how `export` is commented.

Comment (shubhambhokare1) [reply torch/onnx/utils.py:486]: Do I need to add comments in the __init__ file as well?

Comment (BowenBao) [reply torch/onnx/utils.py:486]: Yes, let's put the comments in init file.

**Comment by BowenBao** [comment]
You need to rebase with master to resolve the CI issues.

**Review by BowenBao — DISMISSED** [review]
LGTM

**Review by BowenBao — COMMENTED** [review]
Can you add a test example in `test_utility_funs.py`?

**Review by BowenBao — APPROVED** [review]
LGTM, thanks!

**Review by bzinodev — APPROVED** [review]

**Pull request closed without merging**